### PR TITLE
Ruby 2.7

### PR DESCRIPTION
**Pull request by amichal**
Move us to ruby 2.7 with minimal dependency changes.
- [x] RUBY_VERSION changed from `2.6.5` to `2.7.2` in `.ruby-version,` `docker-compose.yml`, `ci.yml` and production Docker base images
- [x] BUNDLER_VERSION changed from `1.17.3` to `2.2.4` in the same places
- [x] Minor updates to `Gemfile` dependencies to address new warnings and deprecations
- [x] Clean up a few bitrotted comments and a harmless HAML warning
- [x] Switch to scssc-rails for some build spee improvements ?
- [x] Add a .dockerignore so we don't waste bandwidth copying tmp/log etc into build context locally

### CODE REVIEW

**Comment by amichal** [comment]
@yakloinsteak Looks like this is close.  Handing to you to figure out the ECR pre-cache bits so we can try that part out without breaking the 2.6 builds

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+docker
+
+# Dev cruft
+*.swp
+*.swo
+.sass-cache
+yarn-error.log
+.irb-history
+.pry_history
+.pryrc
```
Comment (eanders) [inline .dockerignore:15]: Why are we excluding these?  I find having a synchronized pry history especially useful.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -77,7 +77,7 @@ jobs:
          echo "postgres:5432:*:postgres:postgres" > ~/.pgpass
          chmod 600 ~/.pgpass
 
-         gem install bundler --version=1.17.3
+         gem install bundler --version=2.2.4
```
Comment (eanders) [inline .github/workflows/ci.yml:80]: We've had tons of trouble with bundler 2.x, has that been fixed?

File: Gemfile [hunk 3]
```diff
@@ -253,10 +233,4 @@ group :production, :development, :staging do
   gem 'tiny_tds'
 end
```
Comment (eanders) [inline Gemfile:246]: We should test, but I _think_ removing these will cause issue with deployments, maybe only initial deployments?

**Comment by yakloinsteak** [comment]
@amichal @eanders I've deployed this to QA and things look okay on the surface. In ECR, you'll see images now named with a prefix of "githash" and the pre-cache image with a leading "ruby-2.7.2". @schwartzkylej Does this renaming break any of your monitoring? The lifecycle rules rely on prefix matching, so this lets us clean those rules up and make them much simpler.

@eanders I'll push my workoff jobs, RAM utilization improvements test after this has gone out the door or at least merged into pre-release.

**Comment by schwartzkylej** [comment]
> @schwartzkylej Does this renaming break any of your monitoring? The lifecycle rules rely on prefix matching, so this lets us clean those rules up and make them much simpler.
> 

This change won't break monitoring. Thanks.

**Comment by yakloinsteak** [comment]
@eanders @tquang1708 I pushed a dumb commit that just ignores all of the rubocop problems introduced from merging in pre-release. Is there somebody that has time to work through fixing that? Elliot didn't feel comfortable with `rubocop -a`, but I'm short on time to go through all the changes (though I didn't see any that look scary).

**Comment by eanders** [comment]
@yakloinsteak , I'll review.

**Comment by yakloinsteak** [comment]
@eanders I'm out of time to spend on this.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (amichal) [reply .github/workflows/ci.yml:80]: I did not have any to troubles other than twiddling the timing of "bundler" and the ENV to ensure it was used in all Docker images

File: .dockerignore (hunk 1 above)
Comment (amichal) [reply .dockerignore:15]: We can put them back in. I was excluding  cache, tmp, env-specific like things. Pry just got dragged in without too much thought

**Pull request merged**